Meter.c

```diff
@@ -276,43 +333,915 @@ static void BarMeterMode_draw(Meter* this, int x, int y, int w) {
 
 /* ---------- GraphMeterMode ---------- */
 
-#ifdef HAVE_LIBNCURSESW
+static void GraphMeterMode_reallocateGraphBuffer(Meter* this, const GraphDrawContext* context, size_t nValues) {
+   GraphData* data = &this->drawData;
 
-#define PIXPERROW_UTF8 4
-static const char* const GraphMeterMode_dotsUtf8[] = {
-   /*00*/" ", /*01*/"⢀", /*02*/"⢠", /*03*/"⢰", /*04*/ "⢸",
-   /*10*/"⡀", /*11*/"⣀", /*12*/"⣠", /*13*/"⣰", /*14*/ "⣸",
-   /*20*/"⡄", /*21*/"⣄", /*22*/"⣤", /*23*/"⣴", /*24*/ "⣼",
-   /*30*/"⡆", /*31*/"⣆", /*32*/"⣦", /*33*/"⣶", /*34*/ "⣾",
-   /*40*/"⡇", /*41*/"⣇", /*42*/"⣧", /*43*/"⣷", /*44*/ "⣿"
-};
+   size_t nCellsPerValue = context->nCellsPerValue;
+   size_t valueSize = nCellsPerValue * sizeof(GraphDataCell);
 
-#endif
+   if (!valueSize)
+      goto bufferInitialized;
 
-#define PIXPERROW_ASCII 2
-static const char* const GraphMeterMode_dotsAscii[] = {
-   /*00*/" ", /*01*/".", /*02*/":",
-   /*10*/".", /*11*/".", /*12*/":",
-   /*20*/":", /*21*/":", /*22*/":"
-};
+   data->buffer = xReallocArray(data->buffer, nValues, valueSize);
+
+   // Move existing records ("values") to correct position
+   assert(nValues >= data->nValues);
+   size_t moveOffset = (nValues - data->nValues) * nCellsPerValue;
+   GraphDataCell* dest = (GraphDataCell*)data->buffer + moveOffset;
```

Comment:
Isn't the second line's `+` an implicit multiply by `sizeof(GraphDataCell)`? Is this intended?

Reply:
As you are re-using the pointer in `data->buffer` with that type multiple times below, why now:

```c
size_t moveOffset = (nValues - data->nValues) * nCellsPerValue;
GraphDataCell* cellData = data->buffer;
GraphDataCell* dest = &cellData[moveOffset];
```

Makes the intended pointer arithmetic much more obvious when reading that part.